src/rust/tests/write.rs

```diff
@@ -138,3 +139,45 @@ fn geozero_to_fgb() -> Result<()> {
     fgb.write(&mut fout)?;
     Ok(())
 }
+
+#[test]
+fn test_save_fgb_and_load() {
+    let tempf = tempfile().expect("test failed");
+
+    // Save
+    let linestrings: Vec<LineString<f64>> = vec![
+        geo_types::line_string![
+            (x: -21.95156, y: 64.1446),
+            (x: -21.951, y: 64.14479),
+            (x: -21.95044, y: 64.14527),
+            (x: -21.951445, y: 64.145508)
+        ],
+        geo_types::line_string![(x: 0.0, y: 0.0), (x: 1.0, y: 1.0),],
+    ];
+
+    let mut fgb = FgbWriter::create("test_write", GeometryType::LineString, |_fbb, header| {
+        header.features_count = linestrings.len() as u64;
```

Comment:
This line causes ` failed to fill whole buffer` when reading, because `features_count` is incremented with `add_feature_geom`

Reply:
I have removed the explicit feature count setting.